src/Keboola/CsvMap/Mapper.php

```diff
@@ -70,43 +71,55 @@ private function expandShorthandDefinitions()
         }
     }
 
+    public function parse(array $data, array $userData = [])
+    {
+        // Create a file, even if there is no data
+        $this->getResultFile();
+
+        foreach ($data as $row) {
+            $this->parseRow($row, $userData);
+        }
+    }
+
     /**
-     * @param array $data
+     * @param object|mixed $row
      */
-    public function parse(array $data, array $userData = [])
+    public function parseRow($row, array $userData = [])
     {
         $file = $this->getResultFile();
-        foreach ($data as $row) {
-            $parsedRow = $this->parseRow($row, $userData);
+        $mappedRow = $this->mapRow($row, $userData);
 
-            try {
-                $file->writeRow($parsedRow);
-            } catch (CsvException $e) {
-                if ($e->getCode() != 3) {
-                    throw $e;
-                }
+        try {
+            $file->writeRow($mappedRow);
+        } catch (CsvException $e) {
+            if ($e->getCode() != 3) {
+                throw $e;
+            }
 
-                $columns = [];
-                foreach ($parsedRow as $key => $value) {
-                    if (!is_scalar($value) && !is_null($value)) {
-                        $columns[$key] = gettype($value);
-                    }
+            $columns = [];
+            foreach ($mappedRow as $key => $value) {
+                if (!is_scalar($value) && !is_null($value)) {
+                    $columns[$key] = gettype($value);
                 }
-                $badCols = join(',', array_keys($columns));
-
-                $exception = new BadDataException("Error writing '{$badCols}' column: " . $e->getMessage(), 0, $e);
-                $exception->setData(['bad_columns' => $columns]);
-                throw $exception;
             }
+            $badCols = join(',', array_keys($columns));
+
+            $exception = new BadDataException("Error writing '{$badCols}' column: " . $e->getMessage(), 0, $e);
+            $exception->setData(['bad_columns' => $columns]);
+            throw $exception;
         }
     }
 
-    protected function parseRow($row, array $userData)
+    protected function mapRow($row, array $userData)
     {
         $result = [];
         foreach ($this->mapping as $key => $settings) {
             $delimiter = empty($settings['delimiter']) ? '.' : $settings['delimiter'];
-            $propertyValue = \Keboola\Utils\getDataFromPath($key, $row, $delimiter);
+            // Empty key means "self", ... it is useful to map scalar array values
+            // Eg. row = {"actors":["Patrick Wilson","Rose Byrne","Barbara Hershey"]}
+            //     mapping = {"actors: {"type": "table", "destination": "actor", "tableMapping": {"": "name:} }
+            $propertyValue = $key === '' && is_scalar($row) ? $row : getDataFromPath($key, $row, $delimiter);
```

Comment:
3. `Add the ability to map scalar values from an array to a separated CSV table.`

Pre lepsie pochopenie pozri dokumentaciu:
- https://github.com/keboola/php-csvmap/blob/webrouse-COM-469-enhancements/README.md#map-scalar-items-to-a-separated-csv
- a test: `MapperTest.php` - `testScalarArrayToSeparatedTable`

Pri pisani testov pre `ex-cosmosdb` som zistil, ze neexistuje sposob ako namapovat pole, ktore nie je tvorene objektami ale skalarnymi hodnotami, ... tak som to sem pridal.